### PR TITLE
Login

### PR DESCRIPTION
**Pull request by JamesRexMiller4**
# What's this PR do?

This PR will build out the Login controlled form component, and connect it to the Redux store. Create a fetch POST request from the Form state with error handling. Upon successful login, the Redux store is updated which causes a re-render and redirect back to the home page. 

Update: This PR completes Iteration 2 by adding LogIn and LogOut functionality on buttons, which causes updates to the Redux Store. As well, an error message will conditionally render based on the local state of the Login component if a user enters in incorrect credentials.

## Where should the reviewer start?

The reviewer should start by reviewing the Login component and familiarize themselves with the basic layout and methods that have been added. Next, they should travel to the actions file and review the newly added actions for the Add_User and Update_Logged_In types that will be sent up to the reducers. Then, they should check out the reducers file and review the newly added reducers files isloggedin and user which will handle the newly created actions/action creators. 

Finally they should look at the fetch method that has been created (handleSubmit) as this will take the perform a POST fetch request to the API url, and pass the form's state in the body, and then with the successful response will update the store with the new user info data and change isLoggedIn to true, which will cause the re-render and redirect

Update: They should also review the LogOut functionality

## How should this be manually tested? 
Testing with Redux is currently not in the books, so do not consider any tests. User can manually test this by running npm start, and going through the described process: Arrive to HomePage -> Click LogIn button and be redirected to Login path -> enter in user credentials in form and click submit -> be redirected back to homepage. 

You can check the console to see that the user object is being successfully fetched and in the event that the incorrect credentials are entered they should see the error message handling

### CODE REVIEW

File: rancid-tomatillos/src/Components/Header/Header.js [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
 import './Header.scss';
 import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { removeUser, updateLoggedIn } from '../../actions/index';
 
 
-const Header = ({ data }) => {
+const Header = ({ data, isLoggedIn, removeUser, updateLoggedIn }) => {
+  let log;
+  if (isLoggedIn) {
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/Components/Header/Header.js:10]: Nice work using conditional logic to determine which button is being rendered based on whether or not the user is logged in! the UX/UI is really clean for this.

File: rancid-tomatillos/src/containers/Login/Login.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    super(props);
+    this.state = { 
+      email: '',
+      password: '',
+      error: false 
+     }
+  }
+
+
+  handleChange = (e) => {
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/containers/Login/Login.js:19]: nice work setting up the controlled form!

File: rancid-tomatillos/src/containers/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v1/login', options)
+      .then(res => {
+        if(!res.ok) {
+         throw Error('Incorrect Username/Password')
+        }
+        return res.json()
+      })
+      .then(data => {
+        console.log(data)
+        this.props.addUser(data)
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/containers/Login/Login.js:41]: We could probably deconstruct props at the beginning of this function for greater readability when we use those props

File: rancid-tomatillos/src/containers/Login/Login.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      })
+    .catch(error => {
+      this.setState({error: true})
+    })
+  }
+
+
+
+  render() { 
+    console.log(this.props.loggedInStatus)
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/containers/Login/Login.js:52]: we can get rid of this console log now as we know it is working

File: rancid-tomatillos/src/reducers/user.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export const addUserReducer = (state = {}, action) => {
+  switch (action.type) {
+    case 'ADD_USER':
+      return action.user
+
+    case 'REMOVE_USER':
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/reducers/user.js:6]: good work adding a second case to this reducer!

File: rancid-tomatillos/src/Components/Header/Header.js [hunk 6]
```diff
@@ -1,23 +1,39 @@
...
+    log = <button onClick={() => {updateLoggedIn(isLoggedIn); removeUser();}} className='header-btn'>Log Out</button>
```
Comment (ckoga) [inline rancid-tomatillos/src/Components/Header/Header.js:11]: This is awesome! it nice job working through how to call two functions in an onClick event.

**Review by ckoga — APPROVED** [review]

**Review by adamsjr8576 — COMMENTED** [review]
This all looks really good and the functionality looks really solid and bug free at this point. There were a couple of points where we could deconstruct state/props when accessing those. I think next steps would be to add some styling to the form

**Review by adamsjr8576 — APPROVED** [review]

**Pull request merged**